internal/terraform/ast/variables.go

```diff
@@ -16,13 +16,19 @@ func NewVarsFilename(name string) (VarsFilename, bool) {
 }
 
 func IsVarsFilename(name string) bool {
-	return strings.HasSuffix(name, ".tfvars") && !isIgnoredFile(name)
+	return (strings.HasSuffix(name, ".tfvars") ||
+		strings.HasSuffix(name, ".tfvars.json")) &&
+		!isIgnoredFile(name)
 }
 
 func (vf VarsFilename) String() string {
 	return string(vf)
 }
 
+func (vf VarsFilename) IsJSON() bool {
+	return strings.HasSuffix(string(vf), ".json")
+}
+
 func (vf VarsFilename) IsAutoloaded() bool {
 	name := string(vf)
 	return strings.HasSuffix(name, ".auto.tfvars") || name == "terraform.tfvars"
```

Comment:
I think we'll need to reflect the JSON extensions here as well, otherwise we would never present any diagnostics to the user from JSON tfvars.

For example
https://github.com/hashicorp/terraform-ls/blob/fd0fc089835f3d0c5e23c3ca2a80bc9f27ea2121/internal/langserver/handlers/did_open.go#L81-L84

The high-level idea (reason behind this filtering) is that we present "module-wide" diagnostics more prominently whenever any file from the module is opened - as these are the kind of diagnostics that user would face also when running `validate` or `plan` or `apply` without any additional flags.

The user may also have many individual tfvars files, but we don't present diagnostics from those automatically as we cannot tell which ones will they pick when running validate/plan/apply. Such diagnostics are only presented when these individual files are opened.

Reply:
Oh, I missed that. I extended the check and verified that diagnostics are published.

Reply:
<img width="1205" alt="CleanShot 2021-11-08 at 15 21 34@2x" src="https://user-images.githubusercontent.com/45985/140758592-99bd209a-9eb1-4f8e-b98e-c60d3f4010fc.png">

Reply:
👍🏻  It would be amazing if you could grab a screenshot of this - the diagnostics for `*.tfvars.json` - rendered in the "Problems" pane in VS Code and add it above to symbols as a 2nd use case, but it's not a blocker for merging! 😃